src/browser/fetch/request.rs

```diff
@@ -0,0 +1,282 @@
+    /// The default is `about:client`.
+    pub fn referrer(mut self, referrer: String) -> Self {
+        self.referrer = Some(referrer);
+        self
+    }
+
+    /// Set request referrer policy.
+    pub const fn referrer_policy(mut self, referrer_policy: web_sys::ReferrerPolicy) -> Self {
+        self.referrer_policy = Some(referrer_policy);
+        self
+    }
+
+    /// Set request subresource integrity.
+    pub fn integrity(mut self, integrity: String) -> Self {
+        self.integrity = Some(integrity);
+        self
+    }
+
+    /// Set request timeout in milliseconds.
+    pub fn timeout(mut self, timeout: u32) -> Self {
+        self.timeout = Some(timeout);
+        self
+    }
+}
+
+impl From<Request<'_>> for web_sys::Request {
+    fn from(request: Request) -> Self {
+        let mut init = web_sys::RequestInit::new();
+
+        // headers
+        let headers = web_sys::Headers::new().expect("fetch: cannot create headers");
+        for header in request.headers {
+            headers
+                .append(&header.name, &header.value)
+                .expect("fetch: cannot create header")
+        }
+        init.headers(&headers);
+
+        // method
+        init.method(request.method.as_str());
+
+        // body
+        if let Some(body) = &request.body {
+            init.body(Some(body));
+        }
+
+        // cache
+        if let Some(cache) = request.cache {
+            init.cache(cache);
+        }
+
+        // credentials
+        if let Some(credentials) = request.credentials {
+            init.credentials(credentials);
+        }
+
+        // integrity
+        if let Some(integrity) = &request.integrity {
+            init.integrity(integrity.as_str());
+        }
+
+        // mode
+        if let Some(mode) = request.mode {
+            init.mode(mode);
+        }
+
+        // redirect
+        if let Some(redirect) = request.redirect {
+            init.redirect(redirect);
+        }
+
+        // referrer
+        if let Some(referrer) = &request.referrer {
+            init.referrer(referrer.as_str());
+        }
+
+        // referrer_policy
+        if let Some(referrer_policy) = request.referrer_policy {
+            init.referrer_policy(referrer_policy);
+        }
+
+        // timeout
+        if let Some(timeout) = &request.timeout {
+            let abort_controller = request.controller.clone();
+            *request.controller.timeout_handle.borrow_mut() = Some(
+                // abort request on timeout
+                Timeout::new(*timeout, move || abort_controller.abort()),
+            );
+        }
+
+        // controller
+        // https://developer.mozilla.org/en-US/docs/Web/API/AbortController/signal
+        init.signal(Some(&request.controller.abort_controller.signal()));
+
+        // It seems that the only reason why Request constructor can
+        // fail is when Url contains credentials.  I assume that this
```

Comment:
Could you provide more details? And maybe add a short note to `expect`'s text?

Reply:
Panic looks like this:
```
package.js:280 panicked at 'fetch: Cannot create request: JsValue(TypeError: Failed to construct 'Request': Request cannot be constructed from a URL that includes credentials
```
so it seems that all relevant info is already there.

Reply:
I'm not very happy with it. We are hiding potential error with panic and also breaking `From` trait basic rule. And I can imagine it can fail due to a dozen undocumented reasons in other browsers...